tests/__init__.py

```diff
@@ -71,3 +112,200 @@ def pipeline():
+                            low_rank_background=None,
+                            update_background_components=True,
+                            min_corr=0.8,
+                            min_pnr=10,
+                            normalize_init=False,
+                            center_psf=True,
+                            ssub_B=2,
+                            ring_size_factor=1.4,
+                            del_duplicates=True,
+                            border_pix=0,
+                            min_SNR=3,
+                            rval_thr=0.85,
+                            use_cnn=False)
+
+    miniscope.ProcessingParamSet.insert_new_params(
+        processing_method='caiman',
+        paramset_id=0,
+        paramset_desc='Calcium imaging analysis with CaImAn using default parameters',
+        params=params_caiman)
+
+    yield params_caiman
+
+    if _tear_down:
+        if verbose:
+            (miniscope.ProcessingParamSet & 'paramset_id = 0').delete()
+        else:
+            with QuietStdOut():
+                (miniscope.ProcessingParamSet & 'paramset_id = 0').delete()
+
+
+@pytest.fixture
+def recording_info(pipeline, ingest_sessions):
+    miniscope = pipeline['miniscope']
+
+    miniscope.RecordingInfo.populate()
+
+    yield
+
+    if _tear_down:
+        if verbose:
+            miniscope.RecordingInfo.delete()
+        else:
+            with QuietStdOut():
+                miniscope.RecordingInfo.delete()
+
+
+@pytest.fixture
+def processing_tasks(pipeline, caiman_paramset, recording_info):
+    miniscope = pipeline['miniscope']
+    session = pipeline['session']
+    get_miniscope_root_data_dir = pipeline['get_miniscope_root_data_dir']
+
+    for scan_key in (session.Session * miniscope.Recording - miniscope.ProcessingTask
+                     ).fetch('KEY'):
+        scan_file = find_full_path(get_miniscope_root_data_dir(),
+                                   (miniscope.RecordingInfo.File & scan_key
+                                    ).fetch('file_path')[0])
+
+        recording_dir = scan_file.parent
+        caiman_dir = pathlib.Path(recording_dir / 'caiman')
+        if caiman_dir.exists():
+            miniscope.ProcessingTask.insert1({**scan_key,
+                                              'paramset_id': 0,
+                                              'processing_output_dir': caiman_dir})
+
+    yield
+
+    if _tear_down:
+        if verbose:
+            miniscope.ProcessingTask.delete()
+        else:
+            with QuietStdOut():
+                miniscope.ProcessingTask.delete()
+
+
+@pytest.fixture
+def processing(processing_tasks, pipeline):
+    miniscope = pipeline['miniscope']
+
+    errors = miniscope.Processing.populate(suppress_errors=True)
+
+    if errors:
+        print(f'Populate ERROR: {len(errors)} errors in '
+              + f'"miniscope.Processing.populate()" - {errors[0][-1]}')
+
+    yield
+
+    if _tear_down:
+        if verbose:
```

Comment:
modify this as above with nullcontext to avoid code duplication

Reply:
Thanks for your suggestion. I took this one step further and centralized the check of `verbose` to [one place](https://github.com/CBroz1/workflow-miniscope/blob/dev/tests/__init__.py#L51) to mirror my developments with the deeplabcut workflow